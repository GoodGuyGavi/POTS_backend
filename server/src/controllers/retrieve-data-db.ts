import { IDBModel } from '../commons/types';

const getAllDataDB = (db: IDBModel<any>) => async () => {
	return db.getAll();
};
const getByIDDB = (db: IDBModel<any>) => async id => {
	return db.getById(id);
};

// const getByDeliveryIdDB = (db: IDBModel<any>) => async id => {
//   return db.getByDeliverId(id);
// };

// const getAllByDriverDB = (db: IDBModel<any>) => async id => {
//   return db.getAllByDriver(id);
// };

export { getAllDataDB, getByIDDB };
