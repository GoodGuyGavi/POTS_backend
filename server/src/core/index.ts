import { createUCCreateSupplier } from './_usecases';

export { createUCCreateSupplier };
