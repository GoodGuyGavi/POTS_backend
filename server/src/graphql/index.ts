import { mergeTypes } from 'merge-graphql-schemas';

// import address from "./address";

const typeDefs = [
	//   address,
];

// NOTE: 2nd param is optional, and defaults to false
// Only use if you have defined the same type multiple times in
// different files and wish to attempt merging them together.
export default mergeTypes(typeDefs, { all: true });
