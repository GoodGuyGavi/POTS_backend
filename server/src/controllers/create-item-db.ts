import { IDBModel } from '../commons/types';
import { createUCCreateItem } from '../core/';

const createCreateItemDB = (db: IDBModel<any>) => async i => {
	const createItem = createUCCreateItem();
	const newItem = createItem(i);
	console.log('contorller', newItem);
	return db.insert(newItem);
};

export { createCreateItemDB };
