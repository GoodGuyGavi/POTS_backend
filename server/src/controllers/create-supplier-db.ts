import { IDBModel } from '../commons/types';
import { createUCCreateSupplier } from '../core/_usecases';

const createCreateSupplierDB = (db: IDBModel<any>) => async u => {
	const createSupplier = createUCCreateSupplier();
	const newSupplier = createSupplier(u);
	return db.insert(newSupplier);
};

export { createCreateSupplierDB };
