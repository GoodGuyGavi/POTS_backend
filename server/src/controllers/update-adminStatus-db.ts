import { IDBModel } from '../commons/types';
import { updateUCAdminStatus } from '../core/_usecases';

const updateAdminStatusByIDDB = (db: IDBModel<any>) => async data => {
  const a = await db.getById(data.id);
  const updateAdminStatus = updateUCAdminStatus();
  const newPurchaseOrder = updateAdminStatus(data, a);
  return db.updateById({ ...newPurchaseOrder, id: data.id });
};

export { updateAdminStatusByIDDB };
