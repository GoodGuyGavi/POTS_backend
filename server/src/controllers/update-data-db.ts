import { IDBModel } from '../commons/types';
import { updateUCUser } from '../core/_usecases';

const UpdateByIDDB = (db: IDBModel<any>) => async data => {
	return db.updateById(data);
};

export { UpdateByIDDB };
