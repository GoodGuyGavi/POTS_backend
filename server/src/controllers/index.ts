//User
import { createCreateUserDB } from './create-user-db';
import { DeleteRecordByIDDB } from './delete-data-db';
import { updateUserByIDDB } from './update-user-db';

//Supplier
import { createCreateSupplierDB } from './create-supplier-db';

import { getAllDataDB, getByIDDB } from './retrieve-data-db';

export { getAllDataDB, getByIDDB, createCreateUserDB, DeleteRecordByIDDB, updateUserByIDDB, createCreateSupplierDB };
