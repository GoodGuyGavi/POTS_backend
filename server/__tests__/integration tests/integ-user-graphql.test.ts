import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import mockData from '../mocks/mock-data';
import * as controllers from '../../src/controllers';
import { constructTestServer } from '../__utils';

const {
	//Generic
	getAllDataDB,
	getByIDDB,
	DeleteRecordByIDDB,
	getAllByItemDB,
	getAllByScheduleLineDB,
	getAllBySupplierStatusDB,
	//User
	createCreateUserDB,
	updateUserByIDDB,
} = controllers;

const userMock = {
	insert: jest.fn(async input => {
		return { id: '1', ...input };
	}),
	getById: jest.fn(async id => {
		const filterData = data => {
			if (data.id === id) {
				return data;
			}
		};
		const res = mockData.users.filter(filterData);

		return res[0] || null;
	}),
	getAll: jest.fn(async () => {
		return mockData.users;
	}),
	updateById: jest.fn(async input => {
		return { ...input };
	}),
	deleteById: async id => {
		const filterData = data => {
			if (data.id === id) {
				return data;
			}
		};
		const res = mockData.users.filter(filterData);
		return res[0] || null;
	},
	getAllBySupplierStatus: async id => {},
	getAllByItem: async id => {},
	getAllByScheduleLine: async id => {},
};

const { server }: any = constructTestServer({
	context: {
		createUser: createCreateUserDB(userMock),
		getAllUsers: getAllDataDB(userMock),
		getUserById: getByIDDB(userMock),
		updateUserById: updateUserByIDDB(userMock),
		deleteUserById: DeleteRecordByIDDB(userMock),
	},
});

describe('Tests', () => {
	//Queries
	it('should fetch all user', async () => {
		const USER_ALL = gql`
			query {
				allUsers {
					id
					userName
					password
					userLevel
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({ query: USER_ALL });

		expect(res).toMatchSnapshot();
	});

	it('should fetch one user', async () => {
		const SINGLE_USER = gql`
			query u($id: String!) {
				user(id: $id) {
					id
					userName
					password
					userLevel
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_USER,
			variables: { id: '1' },
		});

		expect(res).toMatchSnapshot();
	});

	it('should error when no user', async () => {
		const SINGLE_USER = gql`
			query u($id: String!) {
				user(id: $id) {
					id
					userName
					password
					userLevel
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_USER,
			variables: { id: '' },
		});

		expect(res).toMatchSnapshot();
	});

	//Mutations
});
