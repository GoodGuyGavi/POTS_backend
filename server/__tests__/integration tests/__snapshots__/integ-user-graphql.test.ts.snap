// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests should error when no user 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch all user 1`] = `
Object {
  "data": Object {
    "allUsers": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field User.id.],
    [GraphQLError: Cannot return null for non-nullable field User.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch one user 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
